src/client/api/i2p_control/session.cc

```diff
@@ -356,184 +192,203 @@ std::string I2PControlSession::GenerateToken() const {
+            break;
+
+          case RouterInfo::NetStatus:
+            response->SetParam(
+                option, static_cast<std::size_t>(kovri::context.GetStatus()));
+            break;
+
+          case RouterInfo::TunnelsParticipating:
+            response->SetParam(
+                option, core::tunnels.GetTransitTunnels().size());
+            break;
+
+          case RouterInfo::ActivePeers:
+            response->SetParam(option, core::transports.GetPeers().size());
+            break;
+
+
+          case RouterInfo::KnownPeers:
+            response->SetParam(option, core::netdb.GetNumRouters());
+            break;
+
+          // Extra options
+          case RouterInfo::DataPath:
+            response->SetParam(option, core::GetCorePath().string());
+            break;
+
+          case RouterInfo::Floodfills:
+            response->SetParam(option, core::netdb.GetNumFloodfills());
+            break;
+
+          case RouterInfo::LeaseSets:
+            response->SetParam(option, core::netdb.GetNumLeaseSets());
+            break;
+
+          case RouterInfo::TunnelsCreationSuccessRate:
+            response->SetParam(
+                option,
+                static_cast<std::size_t>(
+                    core::tunnels.GetTunnelCreationSuccessRate()));
+            break;
+
+          case RouterInfo::TunnelsInList:
+            HandleTunnelsInList(response);
+            break;
+
+          case RouterInfo::TunnelsOutList:
+            HandleTunnelsOutList(response);
+            break;
+
+          case RouterInfo::BWIn15S:
+          case RouterInfo::BWOut15S:
+          case RouterInfo::FastPeers:
+          case RouterInfo::HighCapacityPeers:
+          case RouterInfo::IsReseeding:
+              // TODO(unassigned): implement these indicators
+          default:
+            throw std::runtime_error("Indicator not implemented");
+        }
     }
-  }
 }
 
 void I2PControlSession::HandleRouterManager(
-    const ptree& pt,
-    Response& response) {
+    const Request& request,
+    Response* response)
+{
   LOG(debug) << "I2PControlSession: " << __func__;
-  for (const auto& pair : pt) {
-    if (pair.first == PARAM_TOKEN)
-      continue;
-    LOG(debug) << pair.first;
-    auto it = m_RouterManagerHandlers.find(pair.first);
-    if (it != m_RouterManagerHandlers.end()) {
-      (this->*(it->second))(response);
-    } else {
-      LOG(error)
-        << "I2PControlSession: " << __func__ << ": unknown request " << pair.first;
-      response.SetError(ErrorCode::e_InvalidRequest);
+  for (const auto& pair : request.GetParams())
+    {
+      if (pair.first == Keys::Token)
+        continue;
+      // Cast sanity checks
+      if ((core::GetType(pair.first) < core::GetType(RouterManager::First))
+          || (core::GetType(pair.first) > core::GetType(RouterManager::Last)))
+        {
+          LOG(error) << "I2PControlSession: request contains invalid key not "
+                        "catched during parse !";
```

Comment:
1. s/catched/caught/
2. I know you like to put spaces before your punctuation so I won't cramp your style 😄